app/helpers/embed_player_helper.rb

```diff
@@ -111,19 +92,83 @@ def embed_player_category_options(podcast, selected)
     options_for_select(opts, selected)
   end
 
-  def embed_player_all_episodes_options(selected)
-    opts = %w[all number].map { |v| [t("helpers.label.podcast_player.episodes_options.#{v}"), v] }
-    options_for_select(opts, selected)
-  end
-
   private
 
-  def embed_params(params, preview = false)
+  def embed_params(options_with_defaults = {})
```

Comment:
This method was confusingly named before, so I hijacked it to generate the embed query params.  The above methods (`_episode_url` / `_podcast_url`) call through to this.